notebooks/223-text-prediction/223-text-prediction.ipynb

```diff
@@ -0,0 +1,559 @@
+    "\n",
+    "The following image illustrates the complete demo pipeline used for generation:\n",
+    "\n",
+    "![image2](https://user-images.githubusercontent.com/91228207/163990722-d2713ede-921e-4594-8b00-8b5c1a4d73b5.jpeg)\n",
+    "\n",
+    "And the following image illustrates the pipeline used for conversation:\n",
+    "\n",
+    "![image2](https://user-images.githubusercontent.com/95569637/226101538-e204aebd-a34f-4c8b-b90c-5363ba41c080.jpeg)\n",
+    "\n",
+    "\n",
+    "For Text Generation, The model input is tokenized text, which serves as the initial condition for text generation. Then, logits from the models' inference results are obtained, and the token with the highest probability is selected using the top-k sampling strategy and joined to the input sequence. This procedure repeats until the end of the sequence token is received or the specified maximum length is reached. After that, tokenized IDs are decoded to text.\n",
+    "\n",
+    "For Conversation, User Input is tokenized with eos_token concatenated in the end. Then, the text gets generated as detailed above. The Generated response is added to the history with the eos_token at the end. Additional user input is added to the history, and the sequence is passed back into the model."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set Notebook Mode"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set conversational = True to use PersonaGPT and have a Conversation\n",
+    "conversational = False\n",
+    "# If not using conversational mode, Set use_gpt2 = True to use GPT2 instead of GPT-Neo\n",
+    "use_gpt2 = False "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The Model\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from transformers import GPTNeoForCausalLM, GPT2TokenizerFast, GPT2Tokenizer, GPT2LMHeadModel\n",
+    "\n",
+    "if conversational:\n",
+    "    pt_model = GPT2LMHeadModel.from_pretrained('af1tang/personaGPT')\n",
+    "    tokenizer = GPT2Tokenizer.from_pretrained('af1tang/personaGPT')\n",
+    "elif use_gpt2:\n",
+    "    pt_model = GPT2LMHeadModel.from_pretrained('gpt2')\n",
+    "    tokenizer = GPT2Tokenizer.from_pretrained('gpt2')\n",
+    "else:\n",
+    "    pt_model = GPTNeoForCausalLM.from_pretrained('EleutherAI/gpt-neo-125M')\n",
+    "    tokenizer = GPT2TokenizerFast.from_pretrained('EleutherAI/gpt-neo-125M')"
+   ]
+  },
```

Comment:
Asking the user to set True or False for each model is not the most user-friendly approach.
A better approach would be to give the user a list of the available models and allow them to choose which one they want to use. 
Request to modify the Notebook mode setting accordingly.

Reply:
@Warlord-K Thanks for considering the changes requested. 
1. When I select PersonaGPT or Provide "2", there is some issues with generating the conversation. Request to please test and fix the same.
2. When a user provides a value other than "1", "2" or "0". Display appropriate messages or restrict the action.
3. There is an error thrown in CI Check due to the code in "Set Notebook Mode". Request to validate and fix the same.

Reply:
Sorry for the mistake, I have fixed the code and added restriction on the model number,Please check if this way of taking user input is alright.

Reply:
Thanks for the quick fix. This looks good and the notebook is predicting sequence for all 3 models. 
However, asking the user to enter a number to select the model may not always be the most user-friendly approach. It may be confusing for some users who are not familiar with the numbering system or the different models available. 
Using ipywidgets or any alternate(ensure that it doesn't break the CI checks) to enable users to select the model is a good suggestion. By using ipywidgets, the user can select the model using a drop-down menu or a slider with a default value, making the process more intuitive and user-friendly.
Please refer to Notebooks to refer to widget samples: 235-controlnet-stable-diffusion /225-stable-diffusion-text-to-image etc.

Reply:
I have added the widget for model selection and incorporated the readme text change, sorry I missed that comment earlier. Should i also add a widget for the conversation part?

Reply:
Really appreciate your effort to consider the suggestions and make the necessary changes.